### PR TITLE
Access Rules update

### PR DESCRIPTION
**Pull request by heylemur**
Summary of changes:
- Removed the granting access examples from the Managing_Accounts folders
- Deleted the Composite_Roles examples
- Added tags to the Managing_Resources examples
- Replaced the Managing_Roles examples with the new
- Added info on migrating to access rules to the Readme files

### CODE REVIEW

**Comment by turner-strongdm** [comment]
Spoke with team. g2g.

**Review by turner-strongdm — APPROVED** [review]
Spoke with team. g2g.

File: README.md [hunk 1]
```diff
@@ -4,8 +4,14 @@ This is the examples repository for the [strongDM Java SDK](https://github.com/s
...
-> This repository is currently under construction. The content is expected to change in the coming weeks.
-
+> To increase flexibility when managing thousands of Resources, Role Grants have
+been deprecated in favor of Access Rules, which allow you to grant access based
+on Resource Tags and Type.
+>
+> Previously, you would grant a Role access to specific resources by ID via Role
+Grants. Now, when using Access Rules, the best practice is to grant Resources access based on Type and Tags.
+>
+>The following examples demonstrate Dynamic Access Rules with Tags and Resource Types, as well as Static Access Rules for backwards compatibility with Role Grants. If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did, you can use Resource IDs directly in Access Rules.
```
Comment (heylemur) [inline README.md:14]:
```suggestion
>The following examples demonstrate Dynamic Access Rules with tags and Resource types, as well as Static Access Rules. If it is _necessary_ to grant access to specific Resources in the same way as Role Grants did, you can use Resource IDs directly in Access Rules.
```

File: README.md [hunk 2]
```diff
@@ -4,8 +4,14 @@ This is the examples repository for the [strongDM Java SDK](https://github.com/s
 
 ---
 > **NOTE:**  
-> This repository is currently under construction. The content is expected to change in the coming weeks.
-
+> To increase flexibility when managing thousands of Resources, Role Grants have
```
Comment (heylemur) [inline README.md:7]:
```suggestion
> To increase flexibility when managing a large volume of Resources, Role Grants have
```

File: README.md [hunk 3]
```diff
@@ -4,8 +4,14 @@ This is the examples repository for the [strongDM Java SDK](https://github.com/s
...
+been deprecated in favor of Access Rules, which allow you to grant access based
+on Resource Tags and Type.
```
Comment (heylemur) [inline README.md:9]:
```suggestion
on Resource tags and type.
```

File: README.md [hunk 4]
```diff
@@ -4,8 +4,14 @@ This is the examples repository for the [strongDM Java SDK](https://github.com/s
...
+>
+> Previously, you would grant a Role access to specific resources by ID via Role
```
Comment (heylemur) [inline README.md:11]:
```suggestion
> Previously, you would grant a Role access to specific Resources by ID via Role
```

File: README.md [hunk 5]
```diff
@@ -4,8 +4,14 @@ This is the examples repository for the [strongDM Java SDK](https://github.com/s
...
+Grants. Now, when using Access Rules, the best practice is to grant Resources access based on Type and Tags.
```
Comment (heylemur) [inline README.md:12]:
```suggestion
Grants. Now, when using Access Rules, the best practice is to give Roles access to Resources based on type and tags.
```

File: 3_managing_roles/create_role/src/CreateRole.java [hunk 6]
```diff
@@ -13,40 +12,78 @@
...
+        AccessRule rule2 = new AccessRule();
+        rule2.setType("postgres");
+        AccessRule rule3 = new AccessRule();
+        rule3.setTags(java.util.Map.of(
+            "env", "staging"
+        ));
+        role.setAccessRules(java.util.List.of(rule2, rule3));
+        role = client.roles().update(role).getRole();
+
+        // The RoleGrants API has been deprecated in favor of Access Rules. When
```
Comment (heylemur) [inline 3_managing_roles/create_role/src/CreateRole.java:60]:
```suggestion
        // The Role Grants API has been deprecated in favor of Access Rules. When
```

File: 3_managing_roles/create_role/src/CreateRole.java [hunk 7]
```diff
@@ -13,40 +12,78 @@
...
+        // using Access Rules, the best practice is to grant resources access
```
Comment (heylemur) [inline 3_managing_roles/create_role/src/CreateRole.java:61]:
```suggestion
        // using Access Rules, the best practice is to grant Resources access
```

File: 3_managing_roles/create_role/src/CreateRole.java [hunk 8]
```diff
@@ -13,40 +12,78 @@
...
+        // based on type and tags. If it is _necessary_ to grant access to
+        // specific resources in the same way as RoleGrants did, you can use
```
Comment (heylemur) [inline 3_managing_roles/create_role/src/CreateRole.java:63]:
```suggestion
        // specific resources in the same way as Role Grants did, you can use
```

File: 3_managing_roles/grant_access/src/GrantAccess.java [hunk 9]
```diff
@@ -13,99 +12,141 @@
...
+        rule2.setType("postgres");
+        AccessRule rule3 = new AccessRule();
+        rule3.setTags(java.util.Map.of(
+            "env", "staging"
+        ));
+        role.setAccessRules(java.util.List.of(rule2, rule3));
+        role = client.roles().update(role).getRole();
+
+        // The RoleGrants API has been deprecated in favor of Access Rules. When
+        // using Access Rules, the best practice is to grant resources access
```
Comment (heylemur) [inline 3_managing_roles/grant_access/src/GrantAccess.java:61]:
```suggestion
        // using Access Rules, the best practice is to grant Resources access
```

File: 3_managing_roles/grant_access/src/GrantAccess.java [hunk 10]
```diff
@@ -13,99 +12,141 @@
...
+        // based on type and tags. If it is _necessary_ to grant access to
+        // specific resources in the same way as RoleGrants did, you can use
```
Comment (heylemur) [inline 3_managing_roles/grant_access/src/GrantAccess.java:63]:
```suggestion
        // specific Resources in the same way as Role Grants did, you can use
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 11]
```diff
@@ -13,105 +12,141 @@
...
+        rule2.setType("postgres");
+        AccessRule rule3 = new AccessRule();
+        rule3.setTags(java.util.Map.of(
+            "env", "staging"
+        ));
+        role.setAccessRules(java.util.List.of(rule2, rule3));
+        role = client.roles().update(role).getRole();
+
+        // The RoleGrants API has been deprecated in favor of Access Rules. When
+        // using Access Rules, the best practice is to grant resources access
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:61]:
```suggestion
        // using Access Rules, the best practice is to grant Resources access
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 12]
```diff
@@ -13,105 +12,141 @@
...
+        // based on type and tags. If it is _necessary_ to grant access to
+        // specific resources in the same way as RoleGrants did, you can use
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:63]:
```suggestion
        // specific Resources in the same way as Role Grants did, you can use
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 13]
```diff
@@ -13,105 +12,141 @@
...
+        // resource IDs directly in Access Rules as shown in the following
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:64]:
```suggestion
        // Resource IDs directly in Access Rules as shown in the following
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 14]
```diff
@@ -13,105 +12,141 @@
...
+        }
+        role.getAccessRules().get(0).getIds().add(resourceId2);
+
+        // Update the Role
+        client.roles().update(role);
+    }
+
+	// Example: Delete Role grant via Access Rules
+    private static void deleteRoleGrantViaAccessRulesExample(Client client) {
+        // Create example resources
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:96]:
```suggestion
        // Create example Resources
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 15]
```diff
@@ -13,105 +12,141 @@
...
-            // Detatch user from role
-            client.accountAttachments().delete(attachment.getId());
-            System.out.println("Successfully deleted account attachment.");
-
-            // Delete grant from role
-            client.roleGrants().delete(roleGrant.getId());
-            System.out.println("Successfully deleted role grant.");
-        } catch (Exception e) {
-            e.printStackTrace();
+        // Remove the ID of the second resource
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:110]:
```suggestion
        // Remove the ID of the second Resource
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 16]
```diff
@@ -13,105 +12,141 @@
...
+        java.util.List<String> ids = role.getAccessRules().get(0).getIds();
+        ids.remove(resourceId2);
+
+        // Update the Role
+        client.roles().update(role);
+    }
+
+	// Example: List Role grants via Access Rules
+    private static void listRoleGrantsViaAccessRulesExample(Client client) {
+        // Create example resources
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:120]:
```suggestion
        // Create example Resources
```

File: 3_managing_roles/revoke_access/src/RevokeAccess.java [hunk 17]
```diff
@@ -13,105 +12,141 @@
...
+
+    // Example: Create a Role with empty Access Rules and return the ID
+    private static String createExampleRole(Client client, java.util.List<AccessRule> ar) {
+        Role role = new Role();
+        role.setName("exampleRole-" + Integer.toString(new java.util.Random().nextInt(10000000)));
+        role.setAccessRules(ar);
+        return client.roles().create(role).getRole().getId();
+    }
+
+    // Example: Create a sample resource and return the ID
```
Comment (heylemur) [inline 3_managing_roles/revoke_access/src/RevokeAccess.java:143]:
```suggestion
    // Example: Create a sample Resource and return the ID
```

**Review by jeffreylees — APPROVED** [review]

**Pull request merged**